### PR TITLE
feat: add some more options

### PR DESCRIPTION
**Pull request by argoyle**

### CODE REVIEW

File: k8s.tf [hunk 1]
```diff
@@ -124,9 +126,12 @@ resource "kops_cluster" "k8s" {
   }
 
   api {
-    load_balancer {
-      type  = "Public"
-      class = "Network"
+    dynamic "load_balancer" {
+      for_each = var.api_loadbalancer ? [{}] : []
```
Comment (argoyle) [inline k8s.tf:130]: Is there a better way to solve this?

File: k8s.tf [hunk 2]
```diff
@@ -161,19 +167,48 @@ resource "kops_cluster" "k8s" {
...
+  dynamic "kubelet" {
+    for_each = var.kubelet_auth_webhook ? [{}] : []
+    content {
+      authentication_token_webhook = true
+      authorization_mode           = "Webhook"
+
+      anonymous_auth {
+        value = false
+      }
+    }
```
Comment (argoyle) [inline k8s.tf:184]: Is there a better way to solve this that doesn't duplicate stuff?

File: k8s.tf [hunk 3]
```diff
@@ -161,19 +167,48 @@ resource "kops_cluster" "k8s" {
...
+  }
+
+  dynamic "node_termination_handler" {
+    for_each = !var.node_termination_handler_sqs ? [{}] : []
+    content {
+      enable_prometheus_metrics         = false
+      enable_scheduled_event_draining   = false
+      enable_spot_interruption_draining = false
+      enabled                           = true
+    }
```
Comment (argoyle) [inline k8s.tf:211]: Is there a better way to solve this that doesn't duplicate stuff?

File: k8s.tf (hunk 2 above)
Comment (argoyle) [reply k8s.tf]: Cleaned it up a bit.

File: k8s.tf (hunk 3 above)
Comment (argoyle) [reply k8s.tf]: Cleaned up this one as well

**Review by peter-svensson — APPROVED** [review]
Some small comments.

File: vars.tf [hunk 4]
```diff
@@ -122,3 +122,51 @@ variable "master_policies" {
...
+variable "aws_oidc_provider" {
+  type        = bool
+  default     = false
+  description = "Enable OIDS provider for IRSA, https://kops.sigs.k8s.io/cluster_spec/#service-account-issuer-discovery-and-aws-iam-roles-for-service-accounts-irsa"
+}
+
+variable "service_account_external_permissions" {
+  type        = any
+  default     = null
+  description = "External permissions for Service Accounts (IRSA), "
```
[the remark this answers is not in the document]
Comment (argoyle) [reply vars.tf]: Pasted in the wrong place. 😊 Fixed!

File: vars.tf [hunk 5]
```diff
@@ -122,3 +122,51 @@ variable "master_policies" {
...
+}
+
+variable "api_loadbalancer" {
+  type        = bool
+  default     = true
+  description = "Should a LoadBalancer be created for the Kubernetes API"
```
[the remark this answers is not in the document]
Comment (argoyle) [reply vars.tf:147]: As far as I understand it Route53 is updated with each master IP instead.

File: locals.tf [hunk 6]
```diff
@@ -13,12 +13,48 @@ locals {
...
+    ]
   }
   master_policies = flatten([
     local.master_policies_aws_loadbalancer,
     local.master_policy_addon_bucket_access,
-    var.master_policies]
+    var.master_policies
+    ]
+  )
+  external_permissions_external_dns = {
```
[the remark this answers is not in the document]
Comment (argoyle) [reply locals.tf:25]: Good point. I'll remove it.

[the remark this answers is not in the document]
Comment (peter-svensson) [reply locals.tf:25]: Would make a good "example" though for the readme?

File: k8s.tf [hunk 7]
```diff
@@ -81,52 +81,40 @@ resource "kops_cluster" "k8s" {
...
+    "0.0.0.0/0"
+  ]
   ssh_access = [
-  "0.0.0.0/0"]
+    "0.0.0.0/0"
+  ]
 
   additional_policies = {
     master = jsonencode(local.master_policies)
+    node   = jsonencode(var.node_policies)
```
[the remark this answers is not in the document]
Comment (argoyle) [reply k8s.tf]: Fixed!

File: locals.tf (hunk 6 above)
[the remark this answers is not in the document]
Comment (argoyle) [reply locals.tf]: True! Fixed!

**Pull request merged**